care/facility/tasks/patient/discharge_report.py

```diff
@@ -1,174 +1,184 @@
+        associating_id=consultation.external_id,
+    ).first()
+
+    if file_db_entry is None:
+        file_db_entry: FileUpload = FileUpload.objects.create(
+            name=f"discharge_summary-{consultation.patient.name}-{currnet_date}.pdf",
+            internal_name=f"{uuid4()}.pdf",
+            file_type=FileUpload.FileType.DISCHARGE_SUMMARY.value,
+            associating_id=consultation.external_id,
         )
-        daily_rounds = DailyRound.objects.filter(consultation=consultation)
-        investigations = InvestigationValue.objects.filter(consultation=consultation.id)
-        investigations = list(
-            filter(
-                lambda inv: inv.value is not None or inv.notes is not None,
-                investigations,
-            )
+
+    data = get_discharge_summary_data(consultation)
+    data["date"] = currnet_date
+
+    with tempfile.NamedTemporaryFile(suffix=".pdf") as file:
+        generate_discharge_summary_pdf(data, file)
+        file_db_entry.put_object(file, ContentType="application/pdf")
+        file_db_entry.upload_completed = True
+        file_db_entry.save()
+
+    return file_db_entry
+
+
+@celery.task()
+def generate_and_upload_discharge_summary_task(consultation_id):
+    file_db_entry = generate_and_upload_discharge_summary(consultation_id)
+    return file_db_entry.id
+
+
+@celery.task()
+def email_discharge_summary(consultation_id, email):
+    summary = (
+        FileUpload.objects.filter(
+            file_type=FileUpload.FileType.DISCHARGE_SUMMARY.value,
+            associating_id=consultation_id,
         )
-    else:
-        consultation = None
-        samples = None
-        daily_rounds = None
-        investigations = None
-    date = make_aware(datetime.datetime.now())
-    disease_status = DiseaseStatusEnum(patient.disease_status).name.capitalize()
-    html_string = render_to_string(
-        "reports/patient_pdf_report.html",
-        {
-            "patient": patient,
-            "samples": samples,
-            "consultation": consultation,
-            "consultations": consultations,
-            "dailyrounds": daily_rounds,
-            "date": date,
-            "diseases": diseases,
-            "investigations": investigations,
-            "disease_status": disease_status,
-        },
-    )
-    filename = str(int(round(time.time() * 1000))) + randomString(10) + ".pdf"
-    bytestring_to_pdf(
-        html_string.encode(),
-        default_storage.open(filename, "w+"),
-        **{
-            "no-margins": None,
-            "disable-gpu": None,
-            "disable-dev-shm-usage": False,
-            "window-size": "2480,3508",
-        },
+        .order_by("-created_date")
+        .first()
     )
-    file = default_storage.open(filename, "rb")
+
+    if (
+        summary is not None
+        and not summary.upload_completed
+        and summary.created_date <= timezone.now() - timedelta(minutes=2)
```

Comment:
should we  maintain time restrictions in a system that is designed to be asynchronous?